eng/scripts/Update-PkgVersion.ps1

```diff
@@ -91,9 +91,15 @@ if ($packageSemVer.HasValidPrereleaseLabel() -ne $true){
 }
 
 if (!$packageOldSemVer.IsPrerelease -and ($packageVersion -ne $NewVersionString)) {
+  $whitespace = $propertyGroup["Version"].PreviousSibling
   if (!$propertyGroup.ApiCompatVersion) {
     $propertyGroup.InsertAfter($csproj.CreateElement("ApiCompatVersion"), $propertyGroup["Version"]) | Out-Null
-    $whitespace = $propertyGroup["Version"].PreviousSibling
+    $propertyGroup.InsertAfter($whitespace.Clone(), $propertyGroup["Version"]) | Out-Null
+  }
+  $ApiCompatVersionComment = "The ApiCompatVersion will be updated automatically and does not need to be modified."
+  if (!($propertyGroup.InnerXml -Match $ApiCompatVersionComment)){
+    $comment = $csproj.CreateComment($ApiCompatVersionComment);
```

Comment:
Will this create duplicate comments if one already existed?

Reply:
Nope, that is the purpose of the line immediately prior.

Reply:
It will potentially add it again if anyone modifies the existing comment. Perhaps it might be better to just find the comment and update the text to to match this in all cases instead of trying to find the exact string.

Reply:
Well that's true, but if someone modifies the comment I don't know that we would want to override what they have updated. Still it seems unlikely that someone would modify an auto-generated comment.

Reply:
I was only thinking about cases where something auto-formats the csproj or something along those lines and spacing or ther things change a little we wouldn't end up replacing the comment. However I guess we can wait and see if we hit any issues like that and then adjust accordingly.